From: David Zafman <dzafman@redhat.com>
Date: Thu, 26 May 2016 22:09:42 -0700
Subject: osd: Fix hang on unfound object after mark_unfound_lost is done

Signed-off-by: David Zafman <dzafman@redhat.com>
(cherry picked from commit 73a27533beba3587f8447b4d41d200427c45042b)

Conflicts:
	src/osd/ReplicatedPG.cc (trivial)

(cherry picked from commit 85b7af9c7dbe7bbc3bdb62231bda8aac5634efaa)

Resolves: rhbz#1414613
---
 src/osd/ReplicatedPG.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/osd/ReplicatedPG.cc b/src/osd/ReplicatedPG.cc
index 3f71019..c43565b 100644
--- a/src/osd/ReplicatedPG.cc
+++ b/src/osd/ReplicatedPG.cc
@@ -9861,6 +9861,7 @@ void ReplicatedPG::mark_all_unfound_lost(
       [=]() {
 	requeue_ops(waiting_for_all_missing);
 	waiting_for_all_missing.clear();
+	requeue_object_waiters(waiting_for_unreadable_object);
 	osd->queue_for_recovery(this);
 
 	stringstream ss;
